### PR TITLE
Tolerate `golangci-lint` deprecation errors

### PR DESCRIPTION
**Pull request by michaelfig**
refs: #8865

## Description

We are upgrading Cosmos SDKs in rapid succession, but now some of our older code uses deprecated patterns.  Our linter reported these as a CI failure, but it is likely more helpful not to fail just because things are deprecated.

With a Github Actions matrix, we run two separate `golangci-lint.yml` jobs: one that hides deprecations, and one that only warns on errors instead of failing.

Drive by: fix `bin/agd` and the integration tests preventing merging this PR.

### CODE REVIEW

File: golang/cosmos/app/export.go [hunk 1]
```diff
@@ -46,7 +46,8 @@ func (app *GaiaApp) ExportAppStateAndValidators(
 
 // prepare for fresh start at zero height
 // NOTE zero height genesis is a temporary feature which will be deprecated
-//      in favour of export at a block height
+//
+//	in favour of export at a block height
```
Comment (JimLarson) [inline golang/cosmos/app/export.go]: Clean up the formatting, and consider godoc-standard function comment formatting.

File: .github/workflows/golangci-lint.yml [hunk 2]
```diff
@@ -18,6 +18,9 @@ jobs:
   golangci:
     name: golangci-lint
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        mode: [no-deprecations, no-failure]
```
Comment (JimLarson) [inline .github/workflows/golangci-lint.yml:23]: Consider if we could simplify by only going with the no-deprecations mode. If we want to see the deprecations we can run golangci-lint manually.

**Review by JimLarson — APPROVED** [review]

File: .github/workflows/golangci-lint.yml (hunk 2 above)
Comment (michaelfig) [reply .github/workflows/golangci-lint.yml:23]: I had that originally, but I didn't want to lose track of the deprecations by requiring additional local steps.  It's cheap enough in CI, and I like being able to keep an eye on them.

**Comment by JimLarson** [comment]
Also referencing #8803 for cleaning up the deprecated API usage.

**Pull request merged**